File: examples/basic.py
"""libiforest from Python (ctypes). Run `make shared` first, then this file."""
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "wrappers"))

import numpy as np
from iforest import IsolationForest

rng = np.random.default_rng(0)
X = np.vstack([rng.standard_normal((500, 4)), [[8, 8, 8, 8]]])   # 500 inliers + 1 outlier

f = IsolationForest(n_trees=100).fit(X)
s = f.score(X)
print(f"inlier mean = {s[:500].mean():.3f}   outlier = {s[-1]:.3f}")
print("anomalies (score > 0.6):", int(f.predict(X, threshold=0.6).sum()))

File: examples/basic.jl
# libiforest from Julia (ccall). Run `make shared` first, then this file.
include(joinpath(@__DIR__, "..", "wrappers", "iforest.jl"))
using .IForest

X = vcat(randn(500, 4), [8.0 8.0 8.0 8.0])    # 500 inliers + 1 outlier

f = IForest.fit(X; n_trees = 100)
s = IForest.score(f, X)
println("inlier mean = ", round(sum(s[1:500]) / 500, digits = 3),
        "   outlier = ", round(s[end], digits = 3))
println("anomalies (score > 0.6): ", sum(IForest.predict(f, X; threshold = 0.6)))
